backend/app/core/events/event_bus.py

```
@@ -39,16 +39,27 @@ def register_global_handler(self, handler_func):
+            except Exception:
+                logger.error(
+                    "Event handler '%s' failed for event %s (type: %s)",
+                    handler.__name__, event.id, event.event_type,
+                    exc_info=True,
+                )
+
         # Call global handlers first
         for handler in self.global_handlers:
             logger.info(f"Calling global handler: {handler.__name__}")
-            asyncio.create_task(handler(event))
+            asyncio.create_task(_safe_call(handler, event))
```

Ruff (0.15.2)

[warning] Store a reference to the return value of `asyncio.create_task`

(RUF006)